rfcs/0001-messages.md

```diff
@@ -0,0 +1,278 @@
+
+The receiver side will validate the header using the same method. Instead of panicking the error can be ignored to don't crash and keep waiting for other (potentially valid) messages.
+
+```rust
+if let Ok(header) = msg.header.validate() {
+    ..
+}
+```
+
+## Serialization/Deserialization
+
+Each message struct needs to serialize to bytes representation before it is sent through the wire and must deserialize to the same struct (round trip) on the receiver side. We use `serde` and macro derivations (`Serialize` and `Deserialize`) to automatically implement where possible.
+
+This will require deriving serde in several types defined in `frost.rs`. 
+Manual implementation of serialization/deserialization will be located at a new mod `src/frost/serialize.rs`.
+
+### Byte order
+
+Each byte chunk specified below is in little-endian order unless is specified otherwise.
+
+### Header
+
+The `Header` part of the message is 4 bytes total:
+
+Bytes | Field name | Data type
+------|------------|-----------
+1     | msg_type   | u8
+1     | version    | u8
+1     | sender     | u8
+1     | receiver   | u8
+
+### Primitive types
+
+`Payload`s use data types that we need to specify first. We have 2 primitive types inside the payload messages:
+
+#### `Scalar`
+
+`Scalar` is a an alias for `jubjub::Fr` and this is a `[u64; 4]` as documented in https://github.com/zkcrypto/jubjub/blob/main/src/fr.rs#L16
+
+#### `AffinePoint`
+
+Much of the math in FROST is done using `jubjub::ExtendedPoint`. This is a structure with 5 `jubjub::Fq`s as defined in https://github.com/zkcrypto/jubjub/blob/main/src/lib.rs#L128-L134
+
+Each `Fq` needed to form a `jubjub::ExtendedPoint` are `Scalar`s of `bls12_381` crate. Scalar here is `[u64; 4]` as documented in https://github.com/zkcrypto/bls12_381/blob/main/src/scalar.rs#L16
+
+For message exchange `jubjub::AffinePoint`s are a better choice as they are shorter in bytes, they are formed of 2 `jubjub::Fq` instead of 5: https://github.com/zkcrypto/jubjub/blob/main/src/lib.rs#L70-L73
+
+Conversion from one type to the other is trivial:
+
+https://docs.rs/jubjub/0.6.0/jubjub/struct.AffinePoint.html#impl-From%3CExtendedPoint%3E
+https://docs.rs/jubjub/0.6.0/jubjub/struct.ExtendedPoint.html#impl-From%3CAffinePoint%3E
+
+### Payload
+
+Payload part of the message is variable in size and depends on message type.
+
+#### `MsgDealerBroadcast`
+
+Bytes  | Field name  | Data type
+-------|-------------|-----------
+256    | secret_key  | Scalar
+512    | commitments | AffinePoint
+256+32 | group_public| (AffinePoint, [u8; 32])
+
+#### `MsgCommitments`
+
+Bytes | Field name | Data type
+------|------------|-----------
+512   | hiding     | AffinePoint
+512   | binding    | AffinePoint
+
+#### `MsgSigningPackage`
+
+Bytes                  | Field name     | Data type
+-----------------------|----------------|-----------
+1                      | participants   | u8
+(1+256+256)*partipants | commitments    | Vec<(u8, AffinePoint, AffinePoint)>
```

Comment:
```suggestion
(1+256+256)*participants | commitments    | Vec<(u8, AffinePoint, AffinePoint)>
```

Reply:
Would it be easier to split this tuple into three named fields?
Or to specify a `Participant` sub-type with three named fields?